core/metacling/src/TCling.cxx

```diff
@@ -8968,6 +8976,9 @@ MethodInfo_t* TCling::MethodInfo_Factory(DeclId_t declid) const
 
 MethodInfo_t* TCling::MethodInfo_FactoryCopy(MethodInfo_t* minfo) const
 {
+   // TODO: Why we need a lock here? It was moved from TFunction::TFunction(const TFunction &orig)
```

Comment:
It would appears it is no longer necessary as the `TClingMethodInfo` constructor takes it (and it's base class' constructor are passive).

Reply:
Yes indeed, the copy constructor is declared implicitly and the only constructor that interacts with the interpreter already takes a lock :+1: